src/rust/ide/view/graph-editor/src/builtin/visualization/java_script/sql.js

```diff
@@ -0,0 +1,274 @@
+loadScript('https://cdnjs.cloudflare.com/ajax/libs/sql-formatter/4.0.2/sql-formatter.min.js')
+
+class SqlVisualization extends Visualization {
+    static inputType = 'Any' // 'Standard.Database.Data.Table.Table | Standard.Database.Data.Column.Column' // TODO change this once sum types are properly supported
+    static label = 'SQL Query'
+
+    constructor(api) {
+        super(api)
+        this.setPreprocessorModule('Standard.Visualization.Sql.Visualization')
+        this.setPreprocessorCode(`x -> here.prepare_visualization x`)
+
+        // mock theme
+        function hash(s) {
+            let sum = 0
+            for (let i = 0; i < s.length; ++i) {
+                sum = (sum + 31 * s.charCodeAt(i)) % 255
+            }
+            return sum
+        }
+        function getColor(name) {
+            return [hash('r' + name), hash('g' + name), hash('b' + name), 1]
+        }
+        this.theme = {
+            getColorForType: getColor,
+            getForegroundColorForType: function (x) {
+                return [255, 255, 255, 1]
+            },
+        }
+    }
+
+    onDataReceived(data) {
+        while (this.dom.firstChild) {
+            this.dom.removeChild(this.dom.lastChild)
+        }
+        let parsedData = data
+        if (typeof data === 'string') {
+            parsedData = JSON.parse(data)
+        }
+
+        const style = `
```

Comment:
Do you need to recreate this string on every data received? Please refacotr all parts of the codebase that are not needed to be recomputed on new data, so this will work faster